pkgs/desktops/plasma-5/breeze-plymouth/default.nix

```diff
@@ -8,37 +8,41 @@
   imagemagick,
   netpbm,
   perl,
-  # these will typically need to be set via an override
-  # in a NixOS context
-  nixosBranding ? false,
-  nixosName ? "NixOS",
-  nixosVersion ? "",
+  logoName ? "nixos",
+  logoFile ? null,
+  osName ? "NixOS",
+  osVersion ? "",
   topColor ? "black",
   bottomColor ? "black"
 }:
 
-let
-  logoName = "nixos";
+let 
+  validColors = [ "black" "cardboard_grey" "charcoal_grey" "icon_blue" "paper_white" "plasma_blue" "neon_blue" "neon_green" ];
 in
+  assert lib.asserts.assertOneOf "topColor" topColor validColors;
+  assert lib.asserts.assertOneOf "bottomColor" bottomColor validColors;
+  
 mkDerivation {
   name = "breeze-plymouth";
   nativeBuildInputs = [ extra-cmake-modules ];
-  buildInputs = [ plymouth ] ++ lib.optionals nixosBranding [ imagemagick netpbm perl ];
+  buildInputs = [ plymouth ] ++ lib.optionals (logoFile != null) [ imagemagick netpbm perl ];
   patches = copyPathsToStore (lib.readPathsFromFile ./. ./series);
-  cmakeFlags = lib.optionals nixosBranding [
-    "-DDISTRO_NAME=${nixosName}"
-    "-DDISTRO_VERSION=${nixosVersion}"
+  cmakeFlags = [
+    "-DDISTRO_NAME=${osName}"
+    "-DDISTRO_VERSION=${osVersion}"
     "-DDISTRO_LOGO=${logoName}"
```

Comment:
We shouldn't set `DISTRO_LOGO` if we aren't providing our own - at the moment it will unconditionally get set to `nixos`, whereas leaving it unset will use the default (i.e. `plasma`, which is also shipped).

Reply:
I could set the default logoName to plasma and override it to nixos innixos/modules/system/boot/plymouth.nix

What do you think?

Reply:
I think not setting it should result in getting the default value as defined by the package. Consider: since the logo name determines where to look for the logo, if they changed the default name then we would start asking for a non-existent logo.

So I suggest just not setting the flag if the argument is null (and have that be the default).

Reply:
True that, I modified the code accordingly.